components/patient-measurements/api/src/main/java/org/phenotips/measurements/internal/AbstractMeasurementHandler.java

```diff
@@ -166,6 +199,51 @@ public void initialize() throws InitializationException
         this.chartConfigurations = this.settingsFactory.loadConfigurationsForMeasurementType(getName());
     }
 
+    @Override
+    public Collection<VocabularyTerm> getAssociatedTerms(Double standardDeviation)
+    {
+        ResourceBundle configuration = ResourceBundle.getBundle("measurementsAssociatedTerms");
+        List<VocabularyTerm> terms = new ArrayList<>();
+
+        if (null == standardDeviation || standardDeviation <= -3) {
+            addResolvedTermsToList(configuration, this.getName(), "extremeBelowNormal", terms);
+        }
+        if (null == standardDeviation || standardDeviation <= -2) {
```

Comment:
You can also remove duplicating calls to `addResolvedTermsToList()` by refactoring this to do one call at the end.